src/Microsoft.AspNet.Server.Testing/Deployers/ApplicationDeployerFactory.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Framework.Logging;
+
+namespace Microsoft.AspNet.Server.Testing
+{
+    /// <summary>
+    /// Factory to create an appropriate deployer based on <see cref="DeploymentParameters"/>.
+    /// </summary>
+    public class ApplicationDeployerFactory
+    {
+        /// <summary>
+        /// Creates a deployer instance based on settings in <see cref="DeploymentParameters"/>.
+        /// </summary>
+        /// <param name="deploymentParameters"></param>
+        /// <param name="logger"></param>
+        /// <returns></returns>
+        public static IApplicationDeployer Create(DeploymentParameters deploymentParameters, ILogger logger)
+        {
+            if (deploymentParameters == null)
+            {
+                throw new ArgumentNullException(nameof(deploymentParameters));
+            }
+
+            if (logger == null)
+            {
+                throw new ArgumentNullException(nameof(logger));
+            }
+
+            if (deploymentParameters.RuntimeFlavor == RuntimeFlavor.Mono)
+            {
+                return new MonoDeployer(deploymentParameters, logger);
+            }
+
+            switch (deploymentParameters.ServerType)
```

Comment:
Remember to file a work item re:
We need to think about extensibility here. E.g. how can I use this to test NOWIN? I think we'll need to change these from enums to constants (strings?) and update the factory to let you register deployers and then look them up by key.
For the NOWIN example, I may register "NOWIN" with a factory for creating SelfHostDeployers that specify the "nowin" command to use at launch. WebListener currently maps to the "web" command by convention, and Kestrel to "Kestrel".